src/song_utils.py

```diff
@@ -8,12 +8,52 @@
 from mutagen.id3 import ID3FileType, PictureType
 from mutagen.ogg import OggFileType
 from mutagen.wave import WAVE
-from nextcord import File
+from nextcord import Attachment, Embed, File, Member, Message, User
+from nextcord.utils import escape_markdown
 from PIL import Image, UnidentifiedImageError
 
 import config
 
 
+def embed_song(
+    submit_message: Message | str,
+    attachment_filepath: str,
+    attachment: Attachment,
+    user: User | Member,
+    random_emoji,
+    jump_url=None,
+) -> Embed:
+    if isinstance(submit_message, Message) is False:
```

Comment:
Instead of depending on the type of submit message and having this weird jump_url optional [parameter,](url) I think the `jump_url` param should be mandatory, and `submit_message` should be renamed to `message_content` and should always be a string. Then when embed_song is called the caller must manually pass in the correct params